#
# This patch is created in-house & not yet submitted upstream
#
--- libssh2-1.9.0/src/misc.c.orig	2020-03-22 22:46:50.631196469 +0000
+++ libssh2-1.9.0/src/misc.c	2020-03-22 22:49:43.144694022 +0000
@@ -701,6 +701,9 @@
 #if defined(HAVE_DECL_SECUREZEROMEMORY) && HAVE_DECL_SECUREZEROMEMORY
     SecureZeroMemory(buf, size);
     (void)memset_libssh; /* Silence unused variable warning */
+#elif defined(HAVE_EXPLICIT_MEMSET)
+    (void)explicit_memset(buf, 0, size);
+    (void)memset_libssh; /* Silence unused variable warning */
 #elif defined(HAVE_MEMSET_S)
     (void)memset_s(buf, size, 0, size);
     (void)memset_libssh; /* Silence unused variable warning */
--- libssh2-1.9.0/src/misc.h.orig	2020-03-26 22:09:28.455390231 +0000
+++ libssh2-1.9.0/src/misc.h	2020-03-26 22:16:24.763260000 +0000
@@ -115,6 +115,9 @@
 #endif
 #endif
 
+#define HAVE_EXPLICIT_MEMSET 1
+#undef HAVE_MEMSET_S
+
 void _libssh2_xor_data(unsigned char *output,
                        const unsigned char *input1,
                        const unsigned char *input2,
